rosbag2_transport/test/rosbag2_transport/test_qos.cpp

```diff
@@ -118,23 +120,24 @@ TEST(TestQoS, translates_bad_infinity_values)
   for (const auto & infinity : bad_infinities) {
     std::ostringstream serialized_profile;
     serialized_profile <<
-      "history: 1\n"
+      "history: system_default\n"
       "depth: 10\n"
-      "reliability: 1\n"
-      "durability: 2\n"
+      "reliability: reliable\n"
+      "durability: volatile\n"
       "deadline:\n"
       "  sec: " << infinity.sec << "\n"
       "  nsec: " << infinity.nsec << "\n"
       "lifespan:\n"
       "  sec: " << infinity.sec << "\n"
       "  nsec: " << infinity.nsec << "\n"
-      "liveliness: 0\n"
+      "liveliness: system_default\n"
       "liveliness_lease_duration:\n"
       "  sec: " << infinity.sec << "\n"
       "  nsec: " << infinity.nsec << "\n"
       "avoid_ros_namespace_conventions: false\n";
     const YAML::Node loaded_node = YAML::Load(serialized_profile.str());
-    const auto deserialized_profile = loaded_node.as<rosbag2_transport::Rosbag2QoS>();
+    const auto deserialized_profile = YAML::decode_for_version<rosbag2_storage::Rosbag2QoS>(
+      loaded_node, 9);
```

Comment:
I am curious if we can keep  `loaded_node.as<rosbag2_transport::Rosbag2QoS>();` to make sure that it works by default without specifying a version number explicitly

Reply:
@MichaelOrlov BTW big doubt I just had: this test file shall be transferred in rosbag2_storage package now, am I right?

Reply:
> I am curious if we can keep `loaded_node.as<rosbag2_transport::Rosbag2QoS>();` to make sure that it works by default without specifying a version number explicitly

It works, but I had to set a default value for version param.